Sources/SwiftSyntax/SyntaxConvenienceMethods.swift

```diff
@@ -0,0 +1,60 @@
+//===- SyntaxConvenienceMethods.swift - Convenience funcs for syntax nodes ===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public extension FloatLiteralExprSyntax {
+  var floatingValue: Double {
+    let floatingDigitsWithoutUnderscores = floatingDigits.text.filter {
+      $0 != "_"
+    }
+    return Double(floatingDigitsWithoutUnderscores)!
+  }
+}
+
+public extension IntegerLiteralExprSyntax {
+  var integerValue: Int {
+    let text = self.digits.text
+    let (prefixLength, radix) = prefixLengthAndRadix(text: text)
+    let digitsStartIndex = text.index(text.startIndex, offsetBy: prefixLength)
+    let textWithoutPrefix = String(text.suffix(from: digitsStartIndex))
```

Comment:
I don’t think you need the `String` initialiser wrapping `text.suffix`. It should be more efficient if the `filter` in the next line directly works on the `Substring` returned by `suffix` instead of copying the substring data to a new `String` buffer.

Reply:
Done.